File: Vitura/Vitura.API/Models/Prescription.cs
using System.Text.Json.Serialization;

namespace Vitura.API.Models;

public class Prescription : BaseModel
{
    [JsonPropertyName("patientId")] 
    public int PatientId { get; set; }

    [JsonPropertyName("drugName")] 
    public string DrugName { get; set; }

    [JsonPropertyName("dosage")] 
    public string Dosage { get; set; }

    [JsonPropertyName("datePrescribed")] 
    public DateOnly DatePrescribed { get; set; }
}

File: vitura-frontend/src/types/Prescription.ts

export interface GetPrescriptionDto {
  id: number,
  patientId: number,
  drugName: string,
  dosage: string,
  datePrescribed: string
}

export interface CreatePrescriptionFormValues {
  drugName: string,
  dosage: string,
  datePrescribed: string,
}

export interface CreatePrescriptionDto extends CreatePrescriptionFormValues {
  patientId: number,
}

